apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/internal/ExecutableValidationScope.kt

```diff
@@ -320,11 +315,9 @@ internal class ExecutableValidationScope(
         ))
       }
     }
-  }
 
-  private fun GQLOperationDefinition.validateDeferDirectives() {
-    deferDirectivePathAndLabels.clear()
-    selectionSet.validateDeferDirectives("")
+    if (issues.isNotEmpty()) return
+    selectionSet.validateDeferDirectives()
```

Comment:
Could you go even one step further and reuse `GQLSelectionSet.validate()` (and `GQLField.validate()`, etc...)? This is where the traversal is happening. Not 100% sure if that's possible but feels like it should be?

Reply:
Hehe I contemplated it without really trying and my first impression was that it would make the code more complex/harder to read - but let me try a bit more so we can know for sure 👍

Reply:
All right in the end it's not bad I think 😊 I had to pass `path` around. 
Also there's a small trick to **not** do the `@defer` specific validation in the case where we come from `GQLFragmentDefinition.validate()` because in that case the path+label unicity doesn't make sense.  LMK what you think.

Reply:
We could pass a `TraversalContext` down the rabbit hole if we want more data but this is all internal API so we can always change later if needed 👍